lib/spack/spack/compilers/msvc.py

```diff
@@ -319,18 +335,14 @@ def fc_version(cls, fc):
         if not sys.platform == "win32":
             return "unknown"
         fc_ver = cls.default_version(fc)
-        avail_fc_version.add(fc_ver)
-        fc_path[fc_ver] = fc
-        if os.getenv("ONEAPI_ROOT"):
-            try:
-                sps = spack.operating_systems.windows_os.WindowsOs().compiler_search_paths
-            except AttributeError:
-                raise SpackError("Windows compiler search paths not established")
-            clp = spack.util.executable.which_string("cl", path=sps)
-            ver = cls.default_version(clp)
-        else:
-            ver = fc_ver
-        return ver
+        _avail_fc_version.add(fc_ver)
+        FC_PATH[fc_ver] = fc
+        try:
+            sps = spack.operating_systems.windows_os.WindowsOs().compiler_search_paths
+        except AttributeError:
+            raise SpackError("Windows compiler search paths not established")
```

Comment:
I think this was to guard against a partially executed include that was occurring at one point, let me investigate further.

Reply:
I agree with the exception, my point was more that the message should either:

* more-clearly explain how the user should fix the problem themselves
* or, more-clearly explain that they should report this to us (e.g. if we expect this exception will never be thrown, and are doing something like an `assert`)

Reply:
Right, it's just been so long since this was added I can't recall what the solution from the user's end should be. I _think_ it's reporting it to us but I can't remember exactly, wanted to be sure.